### PR TITLE
Classic redesign

### PR DESCRIPTION
**Pull request by jawn-smith**
This is done to source the initrd and kernel from the rootfs, removing duplication between the gadget and flash-kernel. This will only work with ubuntu-image >= 3.0.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,6 +39,8 @@ gt = $(and $(call ge,$(1),$(2)),$(call ne,$(1),$(2)))
 KERNEL_FLAVOR := $(if $(call gt,$(SERIES_RELEASE),18.04),raspi,raspi2)
 FIRMWARE_FLAVOR := $(if $(call ge,$(SERIES_RELEASE),22.04),raspi,raspi2)
 
+KERNEL_VERSION := $(shell linux-version list | grep $(KERNEL_FLAVOR) | linux-version sort | tail -1)
```
Comment (mwhudson) [inline Makefile]: isn't this going to be the kernel version installed in the host?

Comment (jawn-smith) [reply Makefile]: Yes, pushed an update to fix.

File: Makefile [hunk 2]
```diff
@@ -219,6 +235,13 @@ device-trees: $(SOURCES_RESTRICTED) $(DESTDIR)/boot-assets
 		-name "*.dtbo" -o -name "overlay_map.dtb") \
 		$(DESTDIR)/boot-assets/overlays/
 
+kernel-initrd: $(SOURCES_MAIN)
+	# Update the kernel version in extra_content.yaml
+	$(eval KERNEL_VERSION := $(shell apt-cache $(APT_OPTIONS_KERNEL) policy linux-raspi | grep Candidate | cut -d " " -f 4 | cut -d "." -f 1-4))
```
Comment (mwhudson) [inline Makefile]: This makes an assumption about the kernel being installed, which is part of the image definition, no? Do gadget paths support globs?

Comment (jawn-smith) [reply Makefile]: Gadget paths do not support globs unfortunately. This does require coordination between ubuntu-image and this gadget, but since the gadget tree is now being built as part of the ubuntu-image state machine, it should be a bit safer.

**Comment by waveform80** [comment]
I've tested this out now, but I'm rather confused:

vmlinuz and initrd.img appear on the boot partition; that's fine. But so do vmlinuz-5.19.0-1002-raspi and initrd.img-5.19.0-1002-raspi also appear on the boot partition? Is this what's meant by "removing duplication between the gadget and flash-kernel"?

My concern is that those files don't normally live there, no package will own them (so nothing will ever prune them as the kernel upgrades), flash-kernel will ignore them (so they're effectively useless), and they take up a significant amount of space on a space-limited partition. I do acknowledge this commit also bumps the size of the boot partition, that doesn't help upgraders.

**Comment by waveform80** [comment]
Okay, this is close enough to go with. I've got one more change I'd like to see, but also a bunch of other changes to merge to handle cloud-init deprecations in kinetic, so I'm going to squash merge this (a lot of the commits are redundant at this point and the final diff is simple enough to be one commit), then I'll merge my stuff on top.

**Comment by waveform80** [comment]
This is now merged and I've made the last changes I wanted to see on the top (I've tested these locally too with a kinetic build; haven't re-done the jammy test build yet but an earlier iteration worked nicely).

I've also pushed my latest changes to the `classic` branch and tested (locally) that the two still merge with zero conflicts (which they do). I haven't executed that merge on this repo though, just in case it steps on anything else you've got in the pipeline.
